backend/FwLite/LcmCrdt/Changes/Entries/AddEntryComponentChange.cs

```diff
@@ -38,19 +38,45 @@ public override async ValueTask<ComplexFormComponent> NewEntity(Commit commit, C
         Sense? componentSense = null;
         if (ComponentSenseId is not null)
             componentSense = await context.GetCurrent<Sense>(ComponentSenseId.Value);
-        return new ComplexFormComponent
+        var shouldBeDeleted = (complexFormEntry?.DeletedAt is not null ||
+                               componentEntry?.DeletedAt is not null ||
+                               (ComponentSenseId.HasValue && componentSense?.DeletedAt is not null));
+        var component = new ComplexFormComponent
         {
             Id = EntityId,
             ComplexFormEntryId = ComplexFormEntryId,
             ComplexFormHeadword = complexFormEntry?.Headword(),
             ComponentEntryId = ComponentEntryId,
             ComponentHeadword = componentEntry?.Headword(),
             ComponentSenseId = ComponentSenseId,
-            DeletedAt = (complexFormEntry?.DeletedAt is not null ||
-                         componentEntry?.DeletedAt is not null ||
-                         (ComponentSenseId.HasValue && componentSense?.DeletedAt is not null))
+            DeletedAt = shouldBeDeleted
                 ? commit.DateTime
                 : (DateTime?)null,
         };
+        if (component.DeletedAt is null && await HasReferenceCycle(component, context)) component.DeletedAt = commit.DateTime;
```

Comment:
This is a total nitpick, but I find having the body on the same line as the condition unnecessarily hard to visually parse. Especially when there are function calls and parentheticals.

Reply:
yeah that's a good point, especially for such a long line.